index.rst

```diff
@@ -13,17 +13,188 @@
+
+**Before:** System-test notebooks address specific data holdings
+
+**After:** Notebooks perform a data discovery step and run on arbitrary holdings and/or opt out of data-holding specific checks.
+
+**Discussion:**
+
+From the beginning we have identified the need to have a small data-set that is available on all deployments to allow system-test notebooks to run everywhere.
+While there is merit to this idea, in practice finding the effort to curate such a careful minimal in size but maximal in utility dataset has been hard to find.
+With the advent of the tutorial-notebooks repo, the requirement for performing substantive computations and/or service load has been eliminated from system-test.
+With the proposed data curation notebooks, that require specific data holdings can live elsewhere.
+To the extent that this is practical, system-test service notebooks should be written with a data discovery or data check step to see what data is available (eg. in terms of available catalogs, tap_schema could be queried first to make sure unavailable catalogs are not being requested).
+In practice this is may be an aspirational requirement.
+
+Branches
+--------
+
+**Before:** Mobu typically runs the prod branch of the notebooks (though this is configurable) with notebooks having to be cherry-picked from main to prod.
+
+**After:** End this madness.
+
+**Discussion:**
+
+The need to maintain two different branches has been eliminated with mobu's ability to easily be configured to run off different branches for cases where it is useful to have an "in-development" version deployed.
+Hence cherry-picking is just annoying with no particular benefit.
+
+Outputs
+-------
+
+**Before:** Notebook contributors need to remember to clear outputs before checking in new versions of the notebooks.
+
+**After:** Have this happen automatically (via pre-commit hook or similar), or at least raise a CI warning if there is output checked in.
+
+**Discussion:**
+
+Having the human remember to clear outputs before saving and checking in is error prone. Even if the notebook ends in clear_outputs(), it still implies it was run to the end before commit.
+Ideally something like https://github.com/srstevenson/nb-clean would be integrated in the development workflow.
+
+This may also be of use to other notebook repo maintainers.
+
+
+Write-Only
+----------
+
+**Before:** Notebooks are checked out write-only in nublado to avoid conflicts
+
+**After:** Tutorial-notebooks continue to be write-only (this has saved a lot of support headaches) but system-test notebooks can be read-write to allow for easier guided troubleshooting.
+
+Directories
+-----------
+
+**Before:** Mobu runs notebooks at the root level of repos [?]
+
+**After:** Indicate to mobu whether to run a subdirectory (through eg. a .run file)
+
+**Discussion:** It would be nice to have a top-level directory structure for multiple runable repos, particularly for the data curation notebooks.
+This is not terribly important.
```

Comment:
Another option would be to have the list of directories in which to run notebooks be part of the mobu configuration. I don't have strong feelings about whether that's better than command files in the repository, but it seems a bit cleaner.

Reply:
The reason I didn't go down that route is because then adding (or excluding) a new directory would involve us (or at least a PR to us), as opposed to being completely data driven (so that for example CST can flag through manipulating the repo). If you really don't have strong feelings I think I would like to stick to letting that be driven by metadata of some sort in the notebook repo itself (with a default that is either run everything unless metadata is present, or run only top level unless metadata is present).

Reply:
I'm fine with that, but I think the chances of CST doing this without extensive interactions with us anyway are low to nonexistent, if for no other reason than that they'll probably not notice mobu is failing or not running those notebooks unless we tell them, so I think we may be adding additional complexity to support a workflow that no one will use without us doing it for them.

Reply:
Addressed here https://github.com/lsst-sqre/sqr-080/commit/ff16546098915b658e7a4ef298b443b0fe518294